### PR TITLE
feat: add custom editor API

### LINKED ISSUE #303: Wysiwyg (rich editor) when editing mode?
Hi! it looks like the next most obvious step right? a WYSIWYG editor when editing the articles? is that an idea that is in your minds already??

Something like https://quilljs.com/ or https://ui.toast.com/tui-editor/?

thanks for the hard work!

### PR DESCRIPTION
**Pull request by mathe42**
This PR is a rebased version of #308 

## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

With
```js
content: {
   editor: '~/path/to/component/editor.vue'
}
```
you can define your own component for editing.

Your component should implement the following:
1. v-model for the markdown code
2. prop isEditing (optional)
3. when the edit is finished your component should emit `endEdit`

Resolves #303 

### Example
https://github.com/mathe42/content-editor

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes (if not applicable, please state why)

### Tests
I think existing tests should be enough. I didn't touch the tests and they didn't break.

### CODE REVIEW

**Comment by mathe42** [comment]
Custom editor example
---

To test this with a custom editor (created by me :D ) follow this steps:

1. checkout my branch
2. run:
```
cd docs
yarn add nuxt-content-editor
```
3. add  the following to the nuxt.config.js:

```js
  content: {
    editor: '../../../node_modules/nuxt-content-editor/editor.vue',
  },
```
4. run `yarn dev`
5. checkout the docs at `localhost:3000`

**Comment by mathe42** [comment]
@benjamincanac This is ready for review.

File: docs/content/en/configuration.md [hunk 1]
```diff
@@ -340,11 +340,35 @@ const parseTxt = file => file.split('\n').map(line => line.trim())
 }
 ```
 
+### `editor`
+
+You can provide a custom Editor for editing your markdownfiles in development mode. Just set the `editor` option to a path to your editor component. The code of the default editor you can find [here](https://github.com/nuxt/content/blob/master/packages/content/templates/editor.vue).
```
Comment (Atinux) [inline docs/content/en/configuration.md:345]:
```suggestion
You can provide a custom editor for editing your markdown files in development. Set the `editor` option to a path to your editor component. The code of the default editor you can find [here](https://github.com/nuxt/content/blob/master/packages/content/templates/editor.vue).
```

File: docs/content/en/configuration.md [hunk 2]
```diff
@@ -340,11 +340,35 @@ const parseTxt = file => file.split('\n').map(line => line.trim())
...
+
+
+```js{}[nuxt.config.js]
+// in Config
```
Comment (Atinux) [inline docs/content/en/configuration.md:349]: This comment is not needed

File: docs/content/en/configuration.md [hunk 3]
```diff
@@ -340,11 +340,35 @@ const parseTxt = file => file.split('\n').map(line => line.trim())
...
+
+{
```
Comment (Atinux) [inline docs/content/en/configuration.md:351]: I would actually display:

```js
content {
  editor: '~/path/to/editor/component.vue'
}
```

File: docs/content/en/configuration.md [hunk 4]
```diff
@@ -340,11 +340,35 @@ const parseTxt = file => file.split('\n').map(line => line.trim())
...
+  editor: '~/path/to/editor/component.vue'
+}
+```
+
+Your component should implement the following 3 things:
```
Comment (Atinux) [inline docs/content/en/configuration.md:356]:
```suggestion
Your component should implement the following:
```

File: packages/content/lib/index.js [hunk 5]
```diff
@@ -208,9 +208,15 @@ module.exports = async function (moduleOptions) {
...
       options: {
-        apiPrefix: options.apiPrefixWithBase
+        apiPrefix: options.apiPrefixWithBase,
+        editor: options.editor
       }
     })
+    // Add dev editor component
+    this.addTemplate({
+      fileName: 'content/editor.vue',
+      src: join(__dirname, '../templates/editor.vue')
```
Comment (Atinux) [inline packages/content/lib/index.js:218]:
```suggestion
      src: join(__dirname, '..', 'templates', 'editor.vue')
```

To respect Windows backslahes

**Comment by mathe42** [comment]
## Todos
- [x] Requestet changes
- [x] Alias resolution
- [ ] testing

### Alias resolution
As I understand you mean pathes like `@/components/editor.vue` or `~/components/editor.vue` this should work out of the box...

### Testing
What has to be tested?
- [ ] editor options works and replaces component
- [ ] editor options works with alias e.g. `~/components/editor.vue`
- [x] default editor works as expected (old tests handle this)

**Comment by mathe42** [comment]
@Atinux
Added some thoughts what we can test. I will try to implement them next week. If I missed something just tell me. 👍

**Comment by mathe42** [comment]
@Atinux added some tests...

**Comment by mathe42** [comment]
@benjamincanac @Atinux is there any progress on the review?

**Comment by Atinux** [comment]
Hi @mathe42 

We were on holidays this week.
Will review the coming week, overall looks great!

**Review by Atinux — APPROVED** [review]

**Pull request merged**